solutions/swb-reference/src/environment/ec2VSCode1710/ec2VSCode1710LaunchSSM.yaml

```diff
@@ -0,0 +1,224 @@
+    allowedPattern: ^[a-zA-Z0-9_\-.]{1,100}$
+  DatasetsBucketArn:
+    type: String
+    description: The ARN of the Datasets bucket in the main account
+    allowedPattern: ^arn:aws:s3:::[a-zA-Z0-9_\-\/]+$
+  MainAccountId:
+    type: String
+    description: The Main Account ID where application is deployed
+    allowedPattern: \d{12}
+  MainAccountRegion:
+    type: String
+    description: The region of application deployment in main account
+    allowedPattern: ^\w{2,3}-\w+-\d{1}$
+  MainAccountKeyArn:
+    type: String
+    description: The ARN of main account bucket encryption key
+    allowedPattern: ^arn:aws:kms:(us(-gov)?|ap|ca|cn|eu|sa)-(central|(north|south)?(east|west)?)-\d:\d{12}:key\/[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$
+mainSteps:
+  - name: Launch
+    action: 'aws:executeAwsApi'
+    maxAttempts: 3
+    timeoutSeconds: 30
+    inputs:
+      Service: servicecatalog
+      Api: ProvisionProduct
+      ProductId: '{{ ProductId }}'
+      ProvisionedProductName: '{{ InstanceName }}'
+      PathId: '{{ PathId }}'
+      ProvisioningArtifactId: '{{ ProvisioningArtifactId }}'
+      ProvisioningParameters:
+        - Key: Namespace
+          Value: '{{ Namespace }}'
+        - Key: AmiId
+          Value: '{{ AmiId }}'
+        - Key: InstanceType
+          Value: '{{ InstanceType }}'
+        - Key: KeyName
+          Value: '{{ KeyName }}'
+        - Key: AccessFromCIDRBlock
+          Value: '{{ CIDR }}'
+        - Key: S3Mounts
+          Value: '{{ S3Mounts }}'
+        - Key: IamPolicyDocument
+          Value: '{{ IamPolicyDocument }}'
+        - Key: VPC
+          Value: '{{ VPC }}'
+        - Key: Subnet
+          Value: '{{ Subnet }}'
+        - Key: ALBSecurityGroup
+          Value: '{{ ALBSecurityGroup }}'
+        - Key: ListenerArn
+          Value: '{{ ListenerArn }}'
+        - Key: ListenerRulePriority
+          Value: '{{ ListenerRulePriority }}'
+        - Key: ApplicationUrl
+          Value: '{{ ApplicationUrl }}'
+        - Key: EnvironmentInstanceFiles
+          Value: '{{ EnvironmentInstanceFiles }}'
+        - Key: EncryptionKeyArn
+          Value: '{{ EncryptionKeyArn }}'
+        - Key: DatasetsBucketArn
+          Value: '{{ DatasetsBucketArn }}'
+        - Key: MainAccountId
+          Value: '{{ MainAccountId }}'
+        - Key: MainAccountRegion
+          Value: '{{ MainAccountRegion }}'
+        - Key: MainAccountKeyArn
+          Value: '{{ MainAccountKeyArn }}'
+      Tags:
+        - Key: Env
+          Value: '{{ EnvId }}'
+    outputs:
+      - Name: ProvisionedProductId
+        Type: String
+        Selector: '$.RecordDetail.ProvisionedProductId'
+      - Name: RecordId
+        Type: String
+        Selector: '$.RecordDetail.RecordId'
+    description: Launch Vscode
+    nextStep: WaitForProvisioning
+  - name: WaitForProvisioning
+    action: 'aws:waitForAwsResourceProperty'
+    timeoutSeconds: 600
```

Comment:
Can you change this to 1200, Same change for termination ssm also. Make sure you do this for both versions

Reply:
made the suggested changes in commit c3ac4a835b3b6886d84d5c843038e0f81376215b